flink-runtime/src/main/java/org/apache/flink/streaming/api/windowing/assigners/WindowAssigner.java

```diff
@@ -64,6 +65,12 @@ public abstract Collection<W> assignWindows(
      */
     public abstract Trigger<T, W> getDefaultTrigger();
 
+    /** Returns the default async trigger associated with this {@code WindowAssigner}. */
+    public AsyncTrigger<T, W> getDefaultAsyncTrigger() {
```

Comment:
Add `@Experimental` or `@internal`

Reply:
WindowAssigner is stateless, I think it is not a must to introudce a async version.

Differently, Trigger is bind with state operation, and I need to introduce an async Trigger.